### PR TITLE
feat!: remove obsolete `TS*` highlight groups

### PR DESCRIPTION
**Pull request by clason**
Treesitter highlight groups are handled by upstream Nvim since 0.8.

BREAKING CHANGE: assign highlights to captures directly (e.g., `hi link @keyword Keyword`).

Drops support for Nvim 0.7

### CODE REVIEW

**Review by vigoux — APPROVED** [review]

**Review by theHamsta — APPROVED** [review]

**Comment by clason** [comment]
Closes #3572

File: lua/nvim-treesitter/highlight.lua [hunk 1]
```diff
@@ -175,104 +41,4 @@ function M.detach(bufnr)
   enable_syntax(bufnr)
 end
 
--- TODO(clason): remove obsolete function after bump to 0.8
-function M.set_custom_captures(captures)
-  for capture, hlgroup in pairs(captures) do
```
Comment (theHamsta) [inline lua/nvim-treesitter/highlight.lua:180]: Removing this probably requires updating the readme

Comment (clason) [reply lua/nvim-treesitter/highlight.lua:180]: Ah, missed that. I'll make a fixup PR.

**Comment by numToStr** [comment]
I am just wondering, now that the `TS*` hl groups are removed, how can we define language specific hl groups like `rustTSType`?

**Comment by clason** [comment]
`@type.rust`

**Comment by shadmansaleh** [comment]
Do we have nvim-0.7 compat tag? I think we should have them when we drop support for nvim versions. I can see there's a 0.5 compat branch but don't see one for 0.6 or 0.7. Was a decision already made not to have them after 0.5 ?

**Comment by clason** [comment]
Yes, the decision has been made to only actively support 
1. the latest Nvim release
2. the latest commit on `master`

While nvim-treesitter (just like treesitter support in general) is still considered experimental, it makes no sense to make stronger compatibility guarantees.

**Comment by oxalica** [comment]
Just breaks the highlighting after a plugin update. :disappointed: 
I really think there should be a deprecation-warning time window before the removal.

**Pull request merged**